valohai_yaml/schema/param-item.yaml

```diff
@@ -78,3 +78,12 @@ properties:
     description: |
       The separator to use when interpolating multiple values into a command parameter.
       Defaults to a comma.
+  widget:
+    type: string
+    default: none
+    enum:
+      - none
+      - sql
+      - datumalias
```

Comment:
Let's not lock the given values down here? It's always a pain in the rear to have to update the whole ecosystem depending on this library.
```suggestion
```

Also, I think we should look into the future a bit and allow this to be an object as well as a string:

```yaml
widget: sql
```
would be equivalent to
```yaml
widget:
  type: sql
```
and in the future each widget might have other settings too?

Parsing would always result in a dict or None, serializing would attempt the shortest form, i.e. if the dict in the Step object has `len(widget) == 1 and widget.get("type")`, then it'd be serialized back to a string.